api/receiver/create_run.go

```diff
@@ -46,6 +48,17 @@ func HandleResultsCreate(a AppEngineAPI, s checks.SuitesAPI, w http.ResponseWrit
 	}
 	testRun.CreatedAt = time.Now()
 
+	if len(testRun.FullRevisionHash) != 40 {
+		http.Error(w, "full_revision_hash must be the full SHA (40 chars)", http.StatusBadRequest)
+		return
+	} else if testRun.Revision != "" && strings.Index(testRun.FullRevisionHash, testRun.Revision) != 0 {
+		http.Error(w,
+			fmt.Sprintf("Mismatch of full_revision_hash and revision fields: %s vs %s", testRun.FullRevisionHash, testRun.Revision),
+			http.StatusBadRequest)
+		return
+	}
+	testRun.Revision = testRun.FullRevisionHash[:10]
```

Comment:
Is this needed? You've already checked the consistency in the if condition above.

Reply:
In the case that it's empty, or a matching substring that isn't 10 chars long, this will correct it.